### PR TITLE
Fix Python device type property for XLA and MSNPU

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

**Comment by asuhan** [comment]
Not sure whether there's a more principled way of doing this, but now that extension tensors exist we should report their device type.

File: torch/csrc/tensor/python_tensor.cpp [hunk 1]
```diff
@@ -392,6 +392,9 @@ Device getDevice(const at::Tensor& tensor) {
   if (tensor.is_cuda()) {
     return at::Device(at::DeviceType::CUDA, tensor.get_device());
   }
+  if (tensor.type_id() == XLATensorId() || tensor.type_id() == MSNPUTensorId()) {
+    return at::Device(computeDeviceType(tensor.type_id()), tensor.get_device());
```
Comment (li-roy) [inline torch/csrc/tensor/python_tensor.cpp]: I feel like maybe we can remove this getDevice function entirely and just replace occurrences with a call to tensor.device().

Comment (asuhan) [reply torch/csrc/tensor/python_tensor.cpp]: Done, I've only found two call sites.

**Review by soumith — APPROVED** [review]
thanks Alex!

**Pull request closed without merging**